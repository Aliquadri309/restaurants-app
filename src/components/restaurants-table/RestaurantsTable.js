import React from "react";
import "./RestaurantsTable.scss";

const RestaurantsTable = ({ list, sortBy, onPrev, onNext }) => {
  return (
    <table className="table">
      <thead>
        <tr>
          <th onClick={() => sortBy("name")}>Name</th>
          <th>City</th>
          <th onClick={() => sortBy("state")}>State</th>
          <th>Phone Number</th>
          <th>Genres</th>
        </tr>
      </thead>
      <tbody>
        {list.map((row) => (
          <tr key={row.id}>
            <td>{row.name}</td>
            <td>{row.city}</td>
            <td>{row.state}</td>
            <td>{row.telephone}</td>
            <td>
              {row.genre.split(",").map((g, i) => (
                <span className="genre" key={i}>
                  {g}
                </span>
              ))}
            </td>
          </tr>
        ))}
        {list.length === 0 && (
          <tr>
            <td colSpan="5" className="no-resultts-text">
              No results found. Please change the filter criteria.
            </td>
          </tr>
        )}
      </tbody>
      <tfoot>
        <tr>
          <td colSpan="5">
            <button className="btn prev-btn" onClick={onPrev}>Prev</button>
            <button className="btn next-btn" onClick={onNext}>Next</button>
          </td>
        </tr>
      </tfoot>
    </table>
  );
};

export default RestaurantsTable;
