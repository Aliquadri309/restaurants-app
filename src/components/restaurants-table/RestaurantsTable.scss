.table {
  width: 100%;
  border-collapse: collapse;
  thead {
    tr {
      th {
        padding: 0.75rem;
        border: 1px solid #ddd;
      }
    }
  }
  tbody {
    tr {
      td {
        padding: 0.75rem;
        border: 1px solid #ddd;

        .genre {
          padding: 0.25rem;
          background-color: #8d6e63;
          color: #fff;
          border-radius: 0.25rem;
          margin: 0 0.25rem;
        }
      }
      .no-resultts-text {
        color: #f44336;
        text-align: center;
        font-weight: 500;
        padding: 1rem;
      }
    }
  }
}

.btn {
  border: 1px solid #ddd;
  padding: 0.5rem 0.75rem;
  border-radius: 0.25rem;
  margin-right: 0.5rem;
  margin-top: 1rem;
  cursor: pointer;
}

.prev-btn {
  background-color: #795548;
  color: #fff;
}

.next-btn {
  background-color: #03a9f4;
  color: #fff;
}
