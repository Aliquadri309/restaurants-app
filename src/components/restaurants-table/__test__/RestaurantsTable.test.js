import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import RestaurantsTable from "../RestaurantsTable";

describe("RestaurantsTable Component", () => {
  it("should render without crash", () => {
    const list = [];
    const sortBy = jest.fn();
    const onPrev = jest.fn();
    const onNext = jest.fn();
    const component = shallow(
      <RestaurantsTable
        list={list}
        sortBy={sortBy}
        onNext={onNext}
        onPrev={onPrev}
      />
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it("should have table rows when list has data", () => {
    const list = [
      {
        id: "f223fdd0-4adc-423e-9747-980a66c256ca",
        name: "Old Hickory Steakhouse",
        address1: "201 Waterfront St",
        city: "Oxon Hill",
        state: "MD",
        zip: "20745",
        lat: "38.782098",
        long: "-77.017492",
        telephone: "(301) 965-4000",
        tags: "Social,Food and Dining,Restaurants,Steakhouses",
        website: "http://www.gaylordnational.com",
        genre: "Steak,American,Contemporary,Seafood,Cafe",
        hours: "Open Daily 5:30 PM-10:00 PM",
        attire: "business casual",
      },
      {
        id: "00b35e1a-82b1-4988-b8b9-6df826db2818",
        name: "Matsuhisa",
        address1: "303 E Main St",
        city: "Aspen",
        state: "CO",
        zip: "81611",
        lat: "39.190723",
        long: "-106.82031",
        telephone: "(970) 544-6628",
        tags:
          "Social,Food and Dining,Restaurants,Japanese,Social,Food and Dining,Restaurants,Sushi",
        website: "http://www.matsuhisaaspen.com",
        genre: "Japanese,Sushi,Asian,Contemporary,Seafood",
        hours: "Open Daily 5:30 PM-9:00 PM",
        attire: "business casual",
      },
    ];
    const sortBy = jest.fn();
    const onPrev = jest.fn();
    const onNext = jest.fn();
    const component = shallow(
      <RestaurantsTable
        list={list}
        sortBy={sortBy}
        onNext={onNext}
        onPrev={onPrev}
      />
    );
    const tbody = component.find("tbody tr");
    expect(tbody.length).toBe(2);
  });
  it("should display no data when empty list passed", () => {
    const list = [];
    const sortBy = jest.fn();
    const onPrev = jest.fn();
    const onNext = jest.fn();
    const component = shallow(
      <RestaurantsTable
        list={list}
        sortBy={sortBy}
        onNext={onNext}
        onPrev={onPrev}
      />
    );
    const tbody = component.find("tbody tr");
    expect(tbody.length).toBe(1);
    const noTextNode = component.find(".no-resultts-text");
    expect(noTextNode).toBeDefined();
    expect(noTextNode.text()).toBe(
      "No results found. Please change the filter criteria."
    );
  });
  it("should call sort method when name , state column clicked", () => {
    const list = [];
    const sortBy = jest.fn();
    const onPrev = jest.fn();
    const onNext = jest.fn();
    const component = shallow(
      <RestaurantsTable
        list={list}
        sortBy={sortBy}
        onNext={onNext}
        onPrev={onPrev}
      />
    );
    const nameColumn = component.find("thead tr th").at(0);
    const stateColumn = component.find("thead tr th").at(2);
    nameColumn.simulate("click");
    expect(sortBy).toHaveBeenCalledWith("name");
    stateColumn.simulate("click");
    expect(sortBy).toHaveBeenCalledWith("state");
  });
});
