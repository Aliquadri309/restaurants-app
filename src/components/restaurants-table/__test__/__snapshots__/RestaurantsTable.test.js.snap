// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantsTable Component should render without crash 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th
        onClick={[Function]}
      >
        Name
      </th>
      <th>
        City
      </th>
      <th
        onClick={[Function]}
      >
        State
      </th>
      <th>
        Phone Number
      </th>
      <th>
        Genres
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        className="no-resultts-text"
        colSpan="5"
      >
        No results found. Please change the filter criteria.
      </td>
    </tr>
  </tbody>
  <tfoot>
    <tr>
      <td
        colSpan="5"
      >
        <button
          className="btn prev-btn"
          onClick={[MockFunction]}
        >
          Prev
        </button>
        <button
          className="btn next-btn"
          onClick={[MockFunction]}
        >
          Next
        </button>
      </td>
    </tr>
  </tfoot>
</table>
`;
